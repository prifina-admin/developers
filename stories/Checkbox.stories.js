import React from "react";
import Checkbox from "../src/Checkbox";

export default { title: "Checkbox" };

var state = { checked: false }

var handleCheckboxChange = event => {
    this.setState({ checked: event.target.checked })
}

export const checkbox = () => (
    <div>
        <label>
            <Checkbox />
            One
        </label>
    </div>
);

checkbox.story = {
    name: "Checkbox",
};

export const checkbox2 = () => (
    <div>
        <label>
            <Checkbox width = {"35px"} height = {"35px"}/>
            Two
        </label>
    </div>
);

checkbox2.story = {
    name: "Checkbox2",
};

export const checkbox3 = () => (
    <div>
        <label>
            <Checkbox width = {"35px"} height = {"35px"} bgColor={"red"} borderColor={"green"} marginLeft={"60px"}/>
            Three
        </label>
    </div>
);

checkbox3.story = {
    name: "Checkbox3",
};