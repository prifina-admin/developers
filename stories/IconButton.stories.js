import React from "react";
import IconButton from "../src/IconButton";
import {tableCogIcon} from "@blend-ui/icons";


export default { title: "IconButton" };
export const iconButton = () => (
    <IconButton iconify={tableCogIcon} buttonText={"iconButton"} />
);

iconButton.story = {
    name: "IconButton",
};
