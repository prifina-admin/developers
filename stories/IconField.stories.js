import React from "react";
import IconField from "../src/IconField";
import Input from "../src/Input";
import { accountIcon, eyeIcon, BlendIcon, lockIcon } from "@blend-ui/icons";

export default { title: "Icon Field" };

export const iconfieldinput = () => (
  <IconField>
    <BlendIcon iconify={accountIcon} />
    <Input placeholder={"Enter value here"} />
  </IconField>
);
iconfieldinput.story = {
  name: "Icon Left Field Input ",
};

export const iconfieldinput2 = () => (
  <IconField>
    <Input placeholder={"Enter value here"} />
    <BlendIcon iconify={eyeIcon} />
  </IconField>
);
iconfieldinput2.story = {
  name: "Icon Right Field Input ",
};

export const iconfieldinput3 = () => (
  <IconField>
    <BlendIcon iconify={accountIcon} />
    <Input placeholder={"Enter value here"} />
    <BlendIcon iconify={eyeIcon} />
  </IconField>
);
iconfieldinput3.story = {
  name: "Icon Both Field Input ",
};

export const iconfieldinputPasswd = () => (
    <IconField>
        <BlendIcon iconify={lockIcon} />
        <Input placeholder={"Enter value here"} type={"password"}/>
        <BlendIcon iconify={eyeIcon} />
    </IconField>
);
iconfieldinputPasswd.story = {
    name: "Icon Both Field Input -PWD",
};
