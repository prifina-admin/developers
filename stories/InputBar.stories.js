import React from "react";
import Input from "../src/InputBar";
export default { title: "Input" };

export const input = () => (
    <div>
        <Input bg={"#FFFFFF"} borderColor={"#D1D6DB"} placeholder={"Password"}/>
    </div>


);

input.story = {
    name: "Input Component",
};