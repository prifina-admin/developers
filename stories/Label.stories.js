import React from "react";
import Label from "../src/Label";

export default { title: "Label" };

export const label = () => (
  <Label pointer nowrap variation={"block"}>
    Something very long text next line here{" "}
  </Label>
);
label.story = {
  name: "Label",
};

export const colorLabel = () => (
    <Label pointer nowrap variation={"block"} color={"textAccent"}>
        Something very long text next line here{" "}
    </Label>
);
colorLabel.story = {
    name: "Label with color",
};