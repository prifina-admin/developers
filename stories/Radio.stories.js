import React from "react";
import Radio from "../src/Radio";
import Label from "../src/Label";

export default { title: "Radio"};

export const defaultRadio = () => (
    <div>
        <Radio id={"default"} />
        <Label htmlFor={"default"} >Default Radio Btn</Label>
    </div>
);
defaultRadio.story = {
    name: "Default Radio Btn"
};

export const customRadio = () => (
    <div>
        <Radio id={"custom1"} />
        <Label htmlFor={"custom1"} color={"textAccent"} fontSize={"xl"}>Custom1 Radio Btn</Label>
    </div>
);
customRadio.story = {
    name: "Custom Radio Btn"
};

export const disabledRadio = () => (
    <div>
        <Radio id={"disabled1"} disabled/>
        <Label htmlFor={"disabled1"}
               opacity={"0.5"}
               cursor={"not-allowed"}
        >Disabled Radio Btn</Label>
    </div>
);
disabledRadio.story = {
    name: "Disabled Radio Btn"
};
