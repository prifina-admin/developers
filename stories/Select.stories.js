import React from "react";
import Select from "../src/Select";

export default { title: "Select" };

const names = ['James', 'Paul', 'John', 'George', 'Ringo'];

export const select = () => (
    <Select borderColor={"yellow"} headerName={"type"} optionNames={names} color={"red"} optionColor={"red"} backgroundColor={"white"} optionBackground={"gray"}/>
);

select.story = {
    name: "Select",
};