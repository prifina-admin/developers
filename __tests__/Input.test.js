import React from "react";
import renderer from "react-test-renderer";
import {input} from "../stories/Input.stories";
import Input from "../src/Input";
import {accountIcon, BlendIcon} from "@blend-ui/icons";
import ThemeProvider from "../src/theme/ThemeProvider";

test("Input renders correctly", () => {
    const tree = renderer.create(<ThemeProvider><input type="text" /></ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test("Disabled renders correctly", () => {
    const tree = renderer.create(
        <ThemeProvider>
            <input disabled type="text" />
        </ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test("Input with placeholder renders correctly", () => {
    const tree = renderer.create(
        <ThemeProvider>
            <Input placeholder={"Enter value here"} />
        </ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test("Input Styled no border renders correctly", () => {
    const tree = renderer.create(
        <ThemeProvider>
            <Input placeholder={"Enter value here"} borders={0} p={0} />
        </ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test("Icon Input renders correctly", () => {
    const tree = renderer.create(<BlendIcon iconify={accountIcon} />).toJSON();
    expect(tree).toMatchSnapshot();
});