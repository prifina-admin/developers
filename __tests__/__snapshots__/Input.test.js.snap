// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disabled renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1rk52n4-0 emGfPA"
>
  <input
    disabled={true}
    type="text"
  />
</div>
`;

exports[`Icon Input renders correctly 1`] = `
<svg
  className="Icon-y3r77h-0 hfhfoT"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<path d=\\"M12 4a4 4 0 0 1 4 4a4 4 0 0 1-4 4a4 4 0 0 1-4-4a4 4 0 0 1 4-4m0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4z\\" fill=\\"currentColor\\"/>",
    }
  }
  focusable={false}
  height="24"
  preserveAspectRatio="xMidYMid meet"
  style={
    Object {
      "transform": "rotate(360deg)",
      "verticalAlign": "-0.125em",
    }
  }
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
/>
`;

exports[`Input Styled no border renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1rk52n4-0 emGfPA"
>
  <input
    className="Input__InputElement-sc-1ezb9jk-0 MIWyW"
    placeholder="Enter value here"
    type="text"
    width="100%"
  />
</div>
`;

exports[`Input renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1rk52n4-0 emGfPA"
>
  <input
    type="text"
  />
</div>
`;

exports[`Input with placeholder renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1rk52n4-0 emGfPA"
>
  <input
    className="Input__InputElement-sc-1ezb9jk-0 kBPydZ"
    placeholder="Enter value here"
    type="text"
    width="100%"
  />
</div>
`;
